projects/core/src/routing/configurable-routes/configurable-routes.md

```diff
@@ -0,0 +1,1020 @@
+
+### Extending predefined config
+
+Every part of default config can be extended and overwritten in the Shell App, using `StorefrontModule.withConfig`:
+
+1. for all languages (`default` key):
+
+    ```typescript
+    StorefrontModule.withConfig({
+        routesConfig: {
+            translations: {
+                default: {
+                    product: { paths: ['p/:productCode'] }
+                }
+            }
+        }
+    })
+    ```
+
+    Then the path of the product page will have the shape `p/:productCode` in all languages 
+
+2. for specific languages (for example `en` key):
+
+    ```typescript
+    StorefrontModule.withConfig({
+        routesConfig: {
+            translations: {
+                /* predefined not overwritten: 
+                default: {
+                    product: { paths: ['p/:productCode'] }
+                }
+                */
+                en: {
+                    product: { paths: [':productCode/custom/product-path'] }
+                }
+            }
+        }
+    })
+    ```
+
+    Then the path of the product page will have the shape `:productCode/custom/product-path` only in English. But in every other language it will have the `default` shape `product/:productCode`.
+
+3. both for all languages (`default` key) and for specific languages (for example `en` key):
+
+    ```typescript
+    StorefrontModule.withConfig({
+        routesConfig: {
+            translations: {
+                default: { 
+                    product: { paths: ['p/:productCode'] } // predefined overwritten
+                },
+                en: {
+                    product: { paths: [':productCode/custom/product-path'] }
+                }
+            }
+        }
+    })
+    ```
+
+    Then the path of product page will have shape `:productCode/custom/product-path` only in English. But in every other language it will have the (overwritten) default shape `p/:productCode`.
+
+**How predefined config is extended**
+
+- objects **extend** predefined objects
+- values (primitives, arrays, `null`) **overwrite** predefined values
+
+### Always use params from predefined config
+
+All params that appear in predefined paths (for example `:productCode` param in `product/:productCode` path) mustn't be omitted in overwritten paths. Otherwise Storefront's components may break. For example please **don't do**:
+
+```typescript
+StorefrontModule.withConfig({
+    routesConfig: {
+        translations: {
+            en: {
+                product: { paths: ['product/:productName'] } // overwritten without :productCode
+            }
+        }
+    }
+})
+```
+
+**Routes config can be also fetched from backend**
+
+More in the section [Fetching routes config from backend](#fetching-routes-config-from-backend).
+
+**LIMITATIONS:**
+
+- Only English is supported for now - keys: `en` and `default`. But all languages are planned to be supported.
```

Comment:
When a developer defines translations for `de`, it'll be still a valid config. But it'll be ignored because for now we translate routes only using `en` key - it's hardcoded:
https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/505/files#diff-220b87323fc0b6361de437412586d2b2L13

Translating routes in many languages at once is a future work (it'll require adding prefix to every url - `<base-url>/en/product`, `<base-url>/de/produkt)